src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1047,6 +1031,7 @@ namespace cryptonote
   void core::on_synchronized()
   {
     m_miner.on_synchronized();
+    m_blockchain_storage.safesyncmode();
```

Comment:
on_synchronized is currently called when a peer is synchronized, so this will trigger very early in some cases. I think the best is to check the current blockchain target, and only set safe sync mode is we're at that target already (look for get_target_blockchain_height).